raven/api/raven_channel.py

```diff
@@ -195,3 +196,15 @@ def leave_channel(channel_id):
 		frappe.delete_doc("Raven Channel Member", member.name)
 
 	return "Ok"
+
+
+@frappe.whitelist()
+def mark_all_messages_as_read(channel_ids: list):
+	"""
+	Mark all messages in these channels as read
+	"""    
+	user = frappe.session.user
+	for channel_id in channel_ids:
+		track_channel_visit(channel_id, user=user, publish_event_for_user=True)
```

Comment:
This should just publish one event. The loop will publish multiple times. Call the event publish method after the loop is done.

Reply:
oh okay